internal/dag/builder_test.go

```diff
@@ -1114,7 +1114,7 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 								ListenerName: "ingress_https",
 								routes:       routes(prefixroute("/", service(kuardService))),
 							},
-							Secret: secret(sec1),
+							Secrets: []*Secret{secret(sec1)},
```

Comment:
refactor suggestion: turn this to use a new `secrets()`  helper that takes a variadic list of secrets (the k8s type) and turns it into a list

signature like this: `func secrets(secrets ...*v1.Secret) []*Secret`

should be able to keep the `secret()` helper as well in case it is useful